cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -782,27 +787,27 @@ private void deleteAppMetadata(ApplicationId appId, ApplicationSpecification app
     }
   }
 
-  private Set<ProgramId> getAllPrograms(ApplicationId appId, ApplicationSpecification appSpec) {
+  private Set<ProgramId> getProgramsWithType(ApplicationId appId, ProgramType type,
+                                             Map<String, ? extends ProgramSpecification> programSpec) {
```

Comment:
in general, collections should be named as plural. rename to programSpecs

Reply:
Fixed